app/models/game.rb

```diff
@@ -69,21 +69,227 @@ def player_missing?
   end
 
   def determine_check(king)
-    opponents_pieces = pieces.where(color: !king.color)
+    # if king is in check, returns list of pieces that threaten it - will evaluate to true if present
+    # instead of just returning true/false, I have this method return the list of enemy pieces putting the
+    # king in check to try to DRY up the code
+    threatening_pieces = []
+    opponents_pieces = self.pieces.where(color: !king.color, captured: false)
```

Comment:
added a check to see if pieces are captured to narrow down the piece list and save some processing time